cpp/tests/test_populations.cpp

```diff
@@ -224,3 +225,29 @@ TEST(TestPopulations, set_difference_from_group_total)
         }
     }
 }
+
+TEST(TestPopulations, populations_constraints)
+{
+    // Verify that the functions check_constraints() and apply_constraints() work as expected.
+    mio::Populations<double, InfectionState, AgeGroup> pop(
+        {mio::Index<InfectionState>(InfectionState::Count), mio::Index<AgeGroup>(7)});
+
+    pop.set_group_total<AgeGroup>(mio::Index<AgeGroup>(5), 10);
+    EXPECT_FALSE(pop.check_constraints());
+    EXPECT_FALSE(pop.apply_constraints());
+    EXPECT_NEAR(10, pop.get_total(), 1e-12);
```

Comment:
Please make a brief comment what exactly this tests, and what the expected outcome is (I think "why" should be clear enough in this case).